src/main.js

```diff
@@ -107,10 +102,374 @@ if (WebGL.isWebGLAvailable()) {
+    return renderer;
+}
+
+function setupControls() {
+    controls = new OrbitControls(camera, renderer.domElement);
+    controls.object.up.set(0, 0, 1);
+    controls.update();
+}
+
+function addAxes() {
+    const axesHelper = new THREE.AxesHelper(500);
+    scene.add(axesHelper);
+}
+
+function addAxesLabels() {
+    const labelRenderer = new CSS2DRenderer();
+    labelRenderer.setSize(container.offsetWidth, container.offsetHeight);
+    labelRenderer.domElement.style.position = 'absolute';
+    labelRenderer.domElement.style.pointerEvents = 'none';
+    container.appendChild(labelRenderer.domElement);
+
+    let xLabelDiv = document.createElement('div');
+    xLabelDiv.className = 'text-red-600 text-base p-1 m-1';
+    xLabelDiv.textContent = 'X';
+    let yLabelDiv = document.createElement('div');
+
+    yLabelDiv.className = 'text-axes-green text-base p-1 m-1';
+    yLabelDiv.textContent = 'Y';
+    let zLabelDiv = document.createElement('div');
+
+    zLabelDiv.className = 'text-blue-600 text-base p-1 m-1';
+    zLabelDiv.textContent = 'Z';
+
+    let xLabel = new CSS2DObject(xLabelDiv);
+    let yLabel = new CSS2DObject(yLabelDiv);
+    let zLabel = new CSS2DObject(zLabelDiv);
+
+    const labelOffset = 15;
+    xLabel.position.set(labelOffset, 0, 0);
+    yLabel.position.set(0, labelOffset, 0);
+    zLabel.position.set(0, 0, labelOffset);
+
+    scene.add(xLabel);
+    scene.add(yLabel);
+    scene.add(zLabel);
+    return labelRenderer;
+}
+
+function createCube(location, color, dimensions = new Dimensions(1, 1, 1), opacity = 1) {
+    const geometry = new THREE.BoxGeometry(dimensions.depth, dimensions.width, dimensions.height);
+    const material = new THREE.MeshBasicMaterial({color: color, transparent: true, opacity: opacity});
+    const mesh = new THREE.Mesh(geometry, material);
+    const edges = new THREE.EdgesGeometry(geometry);
+
+    const edgeColor = new THREE.Color(color);
+    edgeColor.offsetHSL(0, 0, 0.2);  // Increase lightness by 20%
+
+    const line = new THREE.LineSegments(edges, new THREE.LineBasicMaterial({color: edgeColor.getHex()}));
+
+    mesh.add(line);
+    mesh.position.set(location.x, location.y, location.z);
+    scene.add(mesh);
+    return mesh;
+}
+
+function changeCubeDimensions(cube, dimensions) {
+    // Remove existing edges
+    cube.children.forEach(child => {
+        if (child instanceof THREE.LineSegments) {
+            cube.remove(child);
+        }
+    });
+
+    const geometry = new THREE.BoxGeometry(dimensions.depth, dimensions.width, dimensions.height);
+    const edges = new THREE.EdgesGeometry(geometry);
+
+    cube.geometry.dispose();
+    cube.geometry = geometry;
+
+    const edgeColor = new THREE.Color(cube.material.color.getHex());
+    edgeColor.offsetHSL(0, 0, 0.2);  // Increase lightness by 20%
+
+    const line = new THREE.LineSegments(edges, new THREE.LineBasicMaterial({ color: edgeColor.getHex() }));
+    cube.add(line);
+}
```

Comment:
`Cube` class?

I'm leaning more and more into the idea that you should create your own utility and setup functions for Three.js. This would make the code feel much more about what you're trying to do rather than about how to interface with Three.js.

I do hesistate in saying that, because it's very easy to get carried away with creating a great wrapper (there are fully fledged libraries that already do that), but I think it would make things more organized and easier to reuse.